### DISCUSSION TITLE
Can I hardcode a set of multiaddresses for a given nodeId in the peerstore ?

### DISCUSSION THREAD

**Discussion by hivenet-mathieu-lacage** [libp2p/go-libp2p#3060] [Q&A] [answered]
Hi,

I'd like to make sure that a couple of peers are always resolved to a set of very specific multiaddrs for api users of the peerstore. The use-case here is for static relays used by autorelay: I would like to make sure that whatever other calls to AddAddrs/SetAddrs, the addresses seen by the addrsplosion.go code are the ones I have hardcoded.

Is there a better way to do this than to re-implement a peerstore with the desired semantics ?

Mathieu

**Comment by sukunrt** [comment]
You can use 
```
	AddAddrs(pid, addrs, peerstore.PermanentAddrTTL)
```
method on the peerstore.

**Reply by hivenet-mathieu-lacage** [reply]
@sukunrt thank you for this feedback and sorry for this late reply. 

Yes, I can do this but it does not ensure that there are no other addresses associated with the peer. i.e., my problem is that my peerstore is polluted with invalid multiaddrs so, I am trying to make sure that whatever any other piece of code that is running and whatever they are doing, they cannot change the nodeid/ip mapping: they cannot remove the only valid one and they cannot add any crap temporary mappings.

Obviously, I also need to fix the code that is adding invalid multiaddrs to the peerstore but that is another story which ties back to the DHT's tendency to take any multiaddr returned from FIND_NODE and stick it in the peerstore temporarily.

**Comment by marten-seemann** [comment, marked as answer]
I don't think libp2p offers a way to do that, but maybe it should?

What you can do though is wrap the libp2p-provided `PeerStore` interface, and implement special logic for those couple of peers.

**Reply by hivenet-mathieu-lacage** [reply]
yes, this is what I ended up doing.
